src/main/java/org/opensearch/flowframework/workflow/CreateIndex/CreateIndexStep.java

```diff
@@ -0,0 +1,115 @@
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow.CreateIndex;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.workflow.WorkflowData;
+import org.opensearch.flowframework.workflow.WorkflowStep;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Step to create an index
+ */
+public class CreateIndexStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(CreateIndexStep.class);
+    private Client client;
+    private final String NAME = "create_index_step";
+
+    /**
+     * Instantiate this class
+     * @param client Client to create an index
+     */
+    public CreateIndexStep(Client client) {
+        this.client = client;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+        CompletableFuture<WorkflowData> future = new CompletableFuture<>();
+        ActionListener<CreateIndexResponse> actionListener = new ActionListener<>() {
+
+            @Override
+            public void onResponse(CreateIndexResponse createIndexResponse) {
+                logger.info("created index:{}", createIndexResponse.index());
+                future.complete(new WorkflowData() {
+                    @Override
+                    public Map<String, Object> getContent() {
+                        return Map.of("index-name", createIndexResponse.index());
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to create an index", e);
+                future.completeExceptionally(e);
+            }
+        };
+
+        String index = null;
+        String type = null;
+        Settings settings = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+            index = (String) content.get("index-name");
+            type = (String) content.get("type");
+            if (index != null && type != null && settings != null) {
+                break;
+            }
+        }
+
+        // TODO:
+        // 1. Create settings based on the index settings received from content
```

Comment:
Since this initial implementation is specifically for a KNN index, we'll need at least a `knn=true` within the index settings.

Reply:
I wouldn't say this is just for k-nn index because the mapping file is for `knn` in this PR. Based on your comment above if the mapping is provided by the user, then we do need to get the settings from the user as well or create dynamic settings based on the mappings provided for any type of index.